### PR TITLE
Add `prefer-method-signature` rule

### PR DESCRIPTION
**Pull request by andy-hanson**
#### PR checklist

- [ ] Addresses an existing issue: #0000
- [X] New feature, bugfix, or enhancement
  - [X] Includes tests
- [X] Documentation update
- [X] Enable CircleCI for your fork (https://circleci.com/add-projects)

#### What changes did you make?

Added the `prefer-method-signature` rule, which recommends to use `foo(): void` over `foo: () => void`.

It looks like these will be semantically equivalent in TS2.2. (And were in 2.0.) https://github.com/Microsoft/TypeScript/issues/13148#issuecomment-270204165

### CODE REVIEW

**Comment by adidahiya** [comment]
How does this interact with `member-ordering`? I believe that rule treats these two different syntaxes differently w/r/t/ ordering.

**Comment by andy-hanson** [comment]
I guess the fix wouldn't fully fix it in that case, because the method would then have to be manually re-ordered.

File: src/rules/preferMethodSignatureRule.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+import * as ts from "typescript";
+
+import * as Lint from "../index";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "prefer-method-signature",
+        description: "Prefer `foo(): void` over `foo: () => void`.",
```
Comment (nchen63) [inline src/rules/preferMethodSignatureRule.ts]: append "in interface and types"

File: src/rules/preferMethodSignatureRule.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        optionsDescription: "Not configurable.",
```
Comment (nchen63) [inline src/rules/preferMethodSignatureRule.ts]: add `hasFix: true,`

**Comment by nchen63** [comment]
@andy-hanson thanks!

**Pull request merged**